### DISCUSSION TITLE
BitTorrent

### DISCUSSION THREAD

**Discussion by anacrolix** [pubky/pkarr#58] [General]
I found this from a comment https://news.ycombinator.com/item?id=39498331. I agree strongly with crotchfire's sentiment about the mainline DHT. I wonder if my project might interest you, it extends a bit beyond using the DHT for DNS but there's some overlap: https://github.com/anacrolix/btlink.

**Comment by Nuhvi** [comment]
@anacrolix Thanks for reaching out. I don't agree with the Hacker News comment regarding why signing the whole packet is bad, but I assume that is not what you care to discuss?

I will read more about your project.

But for now, my general idea is: Mainline is a common, and fragile one even if it is the most robust ever. So I try to use it as little as humanly possible and rely heavily on higher layers of caching on top. I use Mainline only for censorship resistance, and I assume that censorship happens very infrequently and doesn't need to be solved with low latency.

Not sure if your project aligns with that or breaks any of these assumptions, but wanted to share where I am coming from, until I read more about your work.

**Comment by Nuhvi** [comment]
Ok I explored Bitlink a little. Pkarr differs in two ways:
1. It rejects the using of Salts because It wants to scale to give all human beings (and more) censorship resistant IDs, and for that to work, you want as few infohashes per users as possible, salt make it worse, you get multiple infohashes per users (one per subdomain).
2. It uses `.bitlink` TLD, that doesn't sit well with me, because if ICANN decided to make a TLD called `.bitlink`, you will lose. The only winning move here is to use keys as TLDs, ICANN can't take over all keys.
3. Pkarr is not focused on HTTP, rather it focuses on the DNS usecase, one "zonefile" per human, so they can hire and fire any number of cloud hosting providers, so TXT records are just as important and useful, because you can point to your Martrix ID for example, like I do in https://app.pkarr.org/?pk=o4dksfbqk85ogzdb5osziw6befigbuxmuxkuxq8434q89uj56uyy.

So Pkarr is more like a censorship-resistant global phone book, for normal people especially ones that don't care to run their own servers, thus you can see in the main diagram that I suggest that hosting providers are going to republish their own users packets.

**Comment by Nuhvi** [comment]
Finally, some work for enabling serving web pages over Pkarr has been made https://github.com/SeverinAlexB/pkdns using just DNS, which remains lacking the green lock, so I will be interested to see how bitlink solves that and learn from it.

But I don't think I will support using multiple Salts, I considered that while designing Pkarr and decided it is harmful for scalability.

**Reply by anacrolix** [reply]
btlink uses a root certificate with a name constraint that limits it to signing for the `.btlink` subdomain.

**Reply by Nuhvi** [reply]
Is it too hard to do the same for Pkarr TLDs?
